packages/eslint-plugin/src/rules/use-unknown-in-catch-callback-variable.ts

```diff
@@ -235,69 +237,65 @@ export default createRule<[], MessageIds>({
 
     return {
       CallExpression({ arguments: args, callee }): void {
-        const promiseMethodInfo =
-          callee.type === AST_NODE_TYPES.MemberExpression &&
-          tsutils.isThenableType(
-            checker,
-            services.esTreeNodeToTSNodeMap.get(callee),
-            checker.getTypeAtLocation(
-              services.esTreeNodeToTSNodeMap.get(callee.object),
-            ),
-          ) &&
-          [
-            ...[
-              { method: 'catch', append: '' }, // argumentIndexToCheck: 0
-              { method: 'then', append: ' rejection' }, // argumentIndexToCheck: 1
-            ].entries(),
-          ].find(([, { method }]) =>
-            isStaticMemberAccessOfValue(callee, method),
-          );
+        if (callee.type !== AST_NODE_TYPES.MemberExpression) {
+          return;
+        }
+
+        const staticMemberAccessKey = getStaticMemberAccessKey(callee);
+        if (!staticMemberAccessKey) {
+          return;
```

Comment:
Let's get a test case to hit this branch... guessing this will work?
```ts
x[Math.random()];
```
Or anything not statically analyzable 🤷

Reply:
100% coverage 🎉